tests/functional/adapter/test_snapshot.py

```diff
@@ -0,0 +1,291 @@
+"""
```

Comment:
I had to alter the original tests, because UPDATE statements are not allowed on seed tables. So instead, I use different seeds for each stage of the test. We could try to get this changed in dbt-core as well, so that we can use the default test suite again?